### ISSUE TITLE
Upstream to Ubuntu 17.10 and integrate with Ubuntu MATE

### ISSUE THREAD

**Issue by flexiondotorg** [rilian-la-te/vala-panel-appmenu#53]
Hi @rilian-la-te, 

I thought it best to continue this discussion here, which follows on from this:

  * https://github.com/mate-desktop/mate-desktop/pull/256

In Ubuntu MATE 17.40 `mate-desktop` and `mate-settings-daemon` are patched to support Vala Panel AppMenu. I propose that I create a 17.04 (Zesty) PPA for Vala Panel so we can test the integration with Ubuntu MATE and work towards getting Vala Panel in the Debian and Ubuntu archives.

I#ll start by packaging the current Vala Panel release, but I see there have been many fixes recently so you may want to consider a new release :-)

**Comment by rilian-la-te** [comment]
Package a dev-0.4 branch. Master for ValaPanel is broken. Vala Panel AppMenu will be fixed soon, and next I will work for 0.4.

**Comment by rilian-la-te** [comment]
New release of vala-panel-appmenu for you:) Now I will update vala-panel core:)

**Comment by flexiondotorg** [comment]
I'll take a look tomorrow ☺️

**Comment by rilian-la-te** [comment]
Not working, so I hide release for now - found a blocker:( Sorry:( But don't have an idea about debugging it. It requires empty workspace without windows to reproduce.

**Comment by rilian-la-te** [comment]
Not working, so I hide release for now - found a blocker:( Sorry:( But don't have an idea about debugging it. It requires empty workspace without windows to reproduce.

**Comment by rilian-la-te** [comment]
Now you can test it AFAIK.

**Comment by flexiondotorg** [comment]
Working on it...

**Comment by flexiondotorg** [comment]
@rilian-la-te I've updated the `debian/` packaging in my fork:

  * https://github.com/flexiondotorg/vala-panel-appmenu

Sadly `vala-panel-appmenu` fails to build from source (FTBFS).

  * https://launchpadlibrarian.net/314056498/buildlog_ubuntu-zesty-amd64.vala-panel-appmenu_0.3.4-1~zesty1.4_BUILDING.txt.gz

**Comment by rilian-la-te** [comment]
Are you taking GIT or prepared tarball by make package_source? Preparing tarball is important!
Can you try to build it yourself and give log to me? I do not have Ubuntu, sorry:(

**Comment by rilian-la-te** [comment]
If you take zip from git (or remove .git folder before first cmake run), FallbackVersion.cmake will be empty, and package will not be built. I will try to build vala-panel-appmenu in my PPA, If it builds, I will report here.

**Comment by rilian-la-te** [comment]
Packages for Debian Testing I have built successfully (but only with slight modifications and usage of a properly generated tarball)

**Comment by rilian-la-te** [comment]
For zetsy also.
Please, try this: https://github.com/rilian-la-te/vala-panel-appmenu/releases/download/0.3.5/vala-panel-appmenu-0.3.5.tar.gz on the farm.

**Comment by rilian-la-te** [comment]
Try this, please, I built it for Zetsy:
```
sudo add-apt-repository ppa:rilian-la-te/vala-panel-plugins
sudo apt-get update
```

**Comment by flexiondotorg** [comment]
I've built my own version now :-) Next steps are to refine the packaging. In my testing on Ubuntu MATE 17.04 `appmenu-qt5` doesn't seem to be required, do you agree?

Can we drop the `debian/` from your git repository? The reason being, I'd like to get `vala-panel-appmenu` uploaded to Debian (and therefore Ubuntu) and the `debian/` package will need to be maintained in Debian git repository and reference the tarball releases you make.

**Comment by rilian-la-te** [comment]
About appmenu-qt5. Yes, it is not required, but I will recommend to use it if available, because it is more correct implementation of Global Menu than Qt's. 
About dropping debian - if it is required, I will move it to separate repository.
And if you want this in debian, you also must add unity-gtk-module there and use Ubuntu 's patched Gtk or unity-gtk-module with my patch.

**Comment by flexiondotorg** [comment]
That for the info about `appmenu-qt5`, I'll leave it as a `Recommended:` package.

I assumed that `appmenu-qt*` and `unity-gtk*-module` would be in Debian, but they are not. Therefore, I will upload this directly to Ubuntu. I'll file a packaging request and apply to be the maintainer for it with upload rights. 

I suggest you create a repository called `vala-panel-appmenu-packages` and move the `debian/`, `rpm/` and `archlinux/` folders to it. Once the new repository is available I will push my Debian packaging changes to it. My updated Debian packaging can download the source packages you release, so they will be nicely integrated :-)

**Comment by rilian-la-te** [comment]
rpm/ cannot be outside of repository, because spec is generated dynamically. Archlinux and debian, however, will be moved this evening.

**Comment by rilian-la-te** [comment]
Will you package vala-panel itself after it hits 0.4?

**Comment by flexiondotorg** [comment]
Yes, I'm happy to package `vala-panel` too. This needs doing before they are uploaded for the first time ideally. No problem leaving the spec behind.

**Comment by rilian-la-te** [comment]
Ok, I will release 0.3.50 with partial transition. For packaging only:)

**Comment by rilian-la-te** [comment]
1. I have repo for packaging: https://github.com/rilian-la-te/vala-project-packages
2. I have a releases for all 4 products. It builds AFAIK on Arch.
3. libvalapanel cannot be detached from vala-panel now, because 2 versions cannot coexist in same system.
4. Interproject dependencies:
a) Plugins (vala-panel-appmenu and xfce4-sntray-plugin) depends on vala-panel
b) Extras does not depend on anything interproject (AFAIK, you can install one extras and use it anywhere where you can have StatusNotifier)
5. Debian folders is very old and needs updating to latest release.

**Comment by rilian-la-te** [comment]
Hello @flexiondotorg . Can you create package from vala-panel-appmenu-0.4.1 for Debian Sid (with appmenu-gtk-module)? And commit debian folder to  https://github.com/rilian-la-te/vala-project-packages?

**Comment by flexiondotorg** [comment]
I'll take a look, busy this weekend so will do next week.

**Comment by rilian-la-te** [comment]
What debian is missing? 

AFAIK, it builds in sid OK:)

**Comment by amirzolal** [comment]
Even with maximum effort over two days and experimenting with the settings in various .gtkrc or settings.ini files, I haven't found a way to hide the original menu in Mate. This is Mate 1.18 on Ubuntu Mate 17.04. Even built the appmenu from master, still didn't help. The menu works ok, but it's kind of useless when it doesn't dissapear from the GTK3 windows. 

Any idea what will do the trick?

**Comment by rilian-la-te** [comment]
what about gsettings?

**Comment by amirzolal** [comment]
gsettings set  org.gnome.settings-daemon.plugins.xsettings overrides "{'Gtk/ShellShowsAppMenu': <1>, 'Gtk/ShellShowsMenubar': <1>}"
no effect 

gtk-query-settings

           gtk-shell-shows-app-menu: FALSE
            gtk-shell-shows-menubar: FALSE


still, even after the above gsettings command

Any idea?

**Comment by rilian-la-te** [comment]
must be true.
```
[rave@mother ~]$ gsettings get org.mate.interface gtk-shell-shows-app-menu
true
[rave@mother ~]$ gsettings get org.mate.interface gtk-shell-shows-menubar
true
```

**Comment by rilian-la-te** [comment]
set this to true, please.

**Comment by amirzolal** [comment]
Already found it too .. in dconf editor but gsettings works too. 

gsettings set org.mate.interface gtk-shell-shows-app-menu true
gsettings set org.mate.interface gtk-shell-shows-menubar true

now everything works fine
the .gtkrc stuff has no effect

thanks for the help

**Comment by amirzolal** [comment]
Tested with gimp, libreoffice etc. - even there were TopMenu with gnome flashback has problems (missing menu items in LibreOffice sometimes) this works very well. Congratulations.

**Comment by flexiondotorg** [comment]
`vala-panel` and `vala-panel-appmenu` are now available in the Ubuntu 17.10 archive.

**Comment by rilian-la-te** [comment]
thanks. What about sntray-plugin and extras? It also can be used for MATE.
